### PR TITLE
EKS Cloudwatch Logs client

### PR DESCRIPTION
**Pull request by psunnn**
Adding EKSCloudWatchLogs client to access EKS logs.

Note: this PR also modifies the ECSCloudWatchLogs client as the configuration object has changed (log configs are separated by ecs/eks).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,58 +3,86 @@ module github.com/stitchfix/flotilla-os
...
-	github.com/Nvveen/Gotty v0.0.0-20120604004816-cd527374f1e5
-	github.com/Sirupsen/logrus v0.0.0-20170719154753-00386b3fbd63
-	github.com/aws/aws-sdk-go v1.24.0
-	github.com/docker/cli v0.0.0-20170719000837-7f684c751231
-	github.com/docker/distribution v0.0.0-20170718224900-5cfdfbdce59c
-	github.com/docker/docker v0.0.0-20170719201334-8299f1727884
-	github.com/docker/docker-credential-helpers v0.0.0-20170615103903-a8de4f6e8ada
-	github.com/docker/go-connections v0.3.0
-	github.com/docker/go-units v0.3.2
+	github.com/Azure/go-ansiterm v0.0.0-20170929234023-d6e3b3328b78
```
Comment (ujjwalsfix) [inline go.mod]: Azure/MS dependencies?

File: clients/logs/eks_cloudwatch_logs_client.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+	if len(result.Events) == 0 {
+		return "", result.NextForwardToken, nil
+	}
+
+	message := lc.logsToMessage(result.Events)
+	return message, result.NextForwardToken, nil
+}
+
+func (lc *EKSCloudWatchLogsClient) toStreamName(run state.Run) string {
+	return fmt.Sprintf("%s", run.PodName)
```
Comment (ujjwalsfix) [inline clients/logs/eks_cloudwatch_logs_client.go]: Is PodName *string?

File: clients/logs/eks_cloudwatch_logs_client.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	}
+	if !exists {
+		return lc.createNamespace()
+	}
+	return nil
+}
+
+func (lc *EKSCloudWatchLogsClient) namespaceExists() (bool, error) {
+	result, err := lc.logsClient.DescribeLogGroups(&cloudwatchlogs.DescribeLogGroupsInput{
+		LogGroupNamePrefix: &lc.logNamespace,
```
Comment (ujjwalsfix) [inline clients/logs/eks_cloudwatch_logs_client.go]: This behavior probably isn't what we want with EKS, the will be a single namespace. Maybe we dont need the Prefixing?

File: conf/config.yml [hunk 4]
```diff
@@ -69,6 +58,26 @@ http:
...
+  driver:
+    name: awslogs
+    options:
+      awslogs-group: flotilla-os-logs
+      awslogs-stream-prefix: flotilla
+      awslogs-region: us-east-1
+
+eks:
+  log:
+    namespace: flotilla-eks-namespace
```
Comment (ujjwalsfix) [inline conf/config.yml]: What is the purpose of the log-group?

File: flotilla/app.go [hunk 5]
```diff
@@ -70,10 +70,9 @@ func NewApp(conf config.Config,
 	}
 
 	eksLogService, err := services.NewLogService(conf, stateManager, eksLogsClient)
-	// TODO
-	//if err != nil {
-	//	return app, errors.Wrap(err, "problem initializing ecs log service")
-	//}
+	if err != nil {
+		return app, errors.Wrap(err, "problem initializing eks log service")
```
Comment (ujjwalsfix) [inline flotilla/app.go]: do we want an exit 1 if err is not nil?

File: flotilla/endpoints.go [hunk 6]
```diff
@@ -45,8 +44,8 @@ type LaunchRequestV2 struct {
 	Memory           *int64
 	Cpu              *int64
 	Engine           *string
-	NodeLifecycle    *string
-	EphemeralStorage *int64
+	NodeLifecycle    *string `json:"node_lifecycle"`
```
Comment (ujjwalsfix) [inline flotilla/endpoints.go]: Your branch needs to be rebased.

File: go.mod [hunk 7]
```diff
@@ -3,58 +3,86 @@ module github.com/stitchfix/flotilla-os
...
+	google.golang.org/grpc v1.21.0
+	gopkg.in/inf.v0 v0.9.1
 	gopkg.in/tomb.v2 v2.0.0-20161208151619-d5d1b5820637
 	gopkg.in/yaml.v2 v2.2.4
+	honnef.co/go/tools v0.0.0-20190523083050-ea95bdfd59fc // indirect
 	k8s.io/api v0.0.0-20191108065827-59e77acf588f
 	k8s.io/apimachinery v0.0.0-20191108065633-c18f71bf2947
-	k8s.io/client-go v11.0.0+incompatible
-	k8s.io/utils v0.0.0-20191030222137-2b95a09bc58d // indirect
+	k8s.io/client-go v0.0.0-20191108070106-f8f007fd456c
```
Comment (ujjwalsfix) [inline go.mod]: Please rebase these from master

File: vendor/vendor.json [hunk 8]
```diff
@@ -249,10 +249,10 @@
 			"revisionTime": "2019-07-02T19:20:17Z"
 		},
 		{
-			"checksumSHA1": "Q3yvYruVARXNDFcSh5druvyao3U=",
+			"checksumSHA1": "EdsEgZDrBZQ9kjPpvJ1/uvcWK6I=",
```
Comment (ujjwalsfix) [inline vendor/vendor.json]: as mentioned above your branch is not in sync with master.

File: conf/config.yml (hunk 4 above)
Comment (psunnn) [reply conf/config.yml]: Fallback if namespace is not set (per the old ECS cloudwatch client logs)

**Pull request merged**